src/components/calcite-tree/calcite-tree.tsx

```diff
@@ -188,34 +188,42 @@ export class CalciteTree {
         ) as HTMLCalciteTreeItemElement[];
 
         selectedItems.forEach((treeItem) => {
-          if (!targetItems.includes(treeItem)) {
+          if (this.selectedItems.includes(treeItem) && target === treeItem) {
             treeItem.selected = false;
+            this.selectedItems.splice(this.selectedItems.indexOf(treeItem), 1);
           }
         });
       }
-
       if (shouldExpandTarget && !e.detail.forceToggle) {
         target.expanded = true;
       }
 
       if (shouldModifyToCurrentSelection) {
         window.getSelection().removeAllRanges();
       }
-
       if (
         (shouldModifyToCurrentSelection && target.selected) ||
         (shouldSelectChildren && e.detail.forceToggle)
       ) {
-        targetItems.forEach((treeItem) => {
-          treeItem.selected = false;
+        this.selectedItems.forEach((treeItem, i) => {
```

Comment:
Can you simplify this to

```ts
if (this.selectedItems.includes(target)) {
  // deselect `target` and update selected items
}
```
?

This applies to the following `else` statement.